R/SupervisedTask.R

```diff
@@ -1,13 +1,13 @@
-makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data) {
-  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data)
+makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data, spatial) {
+  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data, spatial = spatial)
 
   if (check.data) {
     # costsens does not have a target col...
     # assertCharacter(target, any.missing = FALSE, min.len = 1L)
     w = which.first(target %nin% colnames(data))
     if (length(w) > 0L)
       stopf("Column names of data doesn't contain target var: %s", target[w])
-    checkTaskData(task$env$data, cols = setdiff(colnames(data), target))
+    checkTaskData(task$env$data, cols = setdiff(colnames(task$env$data), target))
```

Comment:
> If you modify the task, the other stuff the user wants to do with it will potentially break. 

Could you expand on this a little? Right now I do not know what the user should want to do with the task which should not be possible anymore. My code only removes `x` and `y` before the training  + predict step in the respective objects created just before these steps. In all other steps, `x` and `y` are still present in `data`. 

If I would put `x` and `y` in `task$task.desc` I would always need to have two objects of `x` and `y` there: (1) which change with the data as it changes (with the additional duplicate subset steps) and (2) backup objects storing the unmodified `x` and `y` objects.

Leaving `x` & `y` at `data` (like now) has the opportunity that this would not be necessary. 

Also, its much more natural to leave `x` and `y` at `data` when doing any other spatial analysis stuff. Forcing the user to separate them from `data` and pass them as an extra (list) object would also cause some confusion I presume. In relation to this, if `x` and `y` are stored with the data, one can use OpenML more easily because the complete dataset is stored there.